File: algorithms/C++/datastructure/array/70.cpp
#include <iostream>

using namespace std;

/**
 * @author: batteria
 * @version: 1.0
 * @since: 2020/12/12
 * @description: 70.climbing-stairs
 */
int climbStairs(int n) {
    if (n <= 2) return n;
    int a = 1, b = 2;
    for (int i = 0; i< n - 2; i++) {
        int t = b;
        b = a + b;
        a = t;
    }
    return b;
}

int main() {
    std::cout << "Hello, World!" << std::endl;
    return 0;
}

File: algorithms/Java/src/algorithm/dynamic_programming/70.java
package algorithm.dynamic_programming;

/**
 * @author: batteria
 * @version: 1.0
 * @since: 2021/3/12
 * @description: 70.爬楼梯
 */
class Solution70 {
    public int climbStairs(int n) {
        if (n == 1) return 1;
        if (n == 2) return 2;
        int a = 1, b = 2;
        for (int i = 0; i < n - 2; i++) {
            int t = b;
            b = a + b;
            a = t;
        }
        return b;
    }

    public static void main(String[] args) {

    }
}
